src/queue.c

```diff
@@ -4144,6 +4142,17 @@ _dispatch_runloop_queue_class_poke(dispatch_queue_t dq)
 		(void)dispatch_assume_zero(kr);
 		break;
 	}
+#elif defined(__linux__)
+	int efd = (int)dq->do_ctxt;
+
+	if (efd != 0) {
```

Comment:
0 is a valid fd, and you will definitely get it if you close STDIN which lots of daemons do, which makes it a bit of an issue. A very gross way to fix that is to store (eventfd + 1) in here